### DISCUSSION TITLE
UPGRADE ECSQL FROM 2.19 TO 3.0.2

### DISCUSSION THREAD

**Discussion by PaoloPetra** [iTwin/itwinjs-core#3466] [Q&A] [answered]
Hi everyone, I'm upgrading my application from 2.19 to 3.0.2
my ECSQL queries that I used seem not to work anymore

SELECT DISTINCT (aspect.ecclassid)
                                 FROM bis.GeometricElement element
                                 JOIN meta.ECClassDef class ON element.ECClassId = class.ECInstanceId
                                 JOIN BisCore.ElementMultiAspect aspect on element.ECInstanceId = aspect.element.id
                                 WHERE class.displaylabel in ('IfcBuilding')
                                 group by aspect.ecclassid

The query returned me the names of the classIds represented in my model

The query still works on the console, but it actually pulls me out now
  "0xa3", "0x1", "0x1ee", etc etc instead of the names as per the query on the console

THANK YOU FOR YOUR VALUABLE HELP

![querySalva](https://user-images.githubusercontent.com/40106886/161541346-1ff953f3-aa22-42db-a12b-c5973ef7a37b.png)

**Comment by ColinKerr** [comment, marked as answer]
ECClassId is a 64 bit Id internally (similar to an Element Id).  In the console we convert the ids to the class names to make it easier to read.

Changes were made to the ECSql result format to provide more performant options and give the caller more control over the results at are returned.  https://www.itwinjs.org/changehistory/3.0.0/#changes-to-ecsql-apis. Changing query row format to use js property names should be the most compatible with the format used in iModel.js: https://www.itwinjs.org/reference/core-common/imodels/queryrowformat/.
